packages/styled-system/src/utils/parse-gradient.ts

```diff
@@ -0,0 +1,70 @@
+import { Dict, memoizedGet as get, isObject } from "@chakra-ui/utils"
+import { ConfigStyle } from "@styled-system/core"
+
+const directionMap = {
+  "to-t": "to top",
+  "to-tr": "to top right",
+  "to-r": "to right",
+  "to-br": "to bottom right",
+  "to-b": "to bottom",
+  "to-bl": "to bottom left",
+  "to-l": "to left",
+  "to-tl": "to top left",
+}
+
+const valueSet = new Set(Object.values(directionMap))
+
+const globals = [
+  "none",
+  "-moz-initial",
+  "inherit",
+  "initial",
+  "revert",
+  "unset",
+]
+
+const trimSpace = (str: string) => str.trim()
+
+export function parseGradient(value: string, theme: Dict) {
+  if (!value || globals.includes(value)) return value
+
+  const regex = /(?<type>^[a-z-A-Z]+)\((?<values>(.*?))\)/g
```

Comment:
IE11 does not support named regex groups, but I am not sure if we should keep supporting / restore support for it.